mlir/include/mlir/Dialect/Linalg/IR/LinalgInterfaces.h

```diff
@@ -30,6 +30,16 @@ class IteratorTypeAttr;
 class LinalgOp;
 class GenericOp;
 
+/// Container for result values of decomposition.
+/// - `decomposedOps` contains operations created by the decomposition that are
+/// returned to the caller for further transformations.
+/// - `decomposedValues` contains the values corresponding to the result of the
+/// aggregate operation.
+struct DecompositionResult {
+  SmallVector<Operation *> decomposedOps;
```

Comment:
Are all decomposed ops Linalg ops? If so, can we use `linalg::LinalgOp` (interface) instead of the generic `Operation *`.

Reply:
In softmax case - yes. I was thinking this could eventually become not a linalg-specific interface, so went with `Operation*`. Do you have a specific reason it should be `LinalgOp` in mind?

Reply:
I think it makes sense to be a bit more restrictive to begin with, then expand as we have more use cases. The main problem is when someone else tries to use it for a wildly different case and complaints "it's not working" when in truth it was never intended to be supported.

Reply:
I'm asking because this seems to be not a trivial change. `LinalgOp` is an incomplete type in the context and a forward declaration for `DecompositionResult` makes the struct an incomplete type for `FailureOr` template instantiation (in `AggregatedOpInterface`).

Reply:
Hm, I see. I'm ok with this being `Operation *` for now, with a TODO/FIXME to make sure people are aware if they try to use it for other needs. I think the probability of anyone trying to use it outside of this scope is very small.

Reply:
Added a todo.

Reply:
I prefer it to be `Operation *`. Thats the more general form, and changing ABI causes breakages. There is no reason for this to be Linalg ops always.